korangar/src/loaders/model/mod.rs

```diff
@@ -39,6 +41,8 @@ impl ModelLoader {
             true => NativeModelVertex::calculate_normal(vertex_positions[0], vertex_positions[1], vertex_positions[2]),
             false => NativeModelVertex::calculate_normal(vertex_positions[2], vertex_positions[1], vertex_positions[0]),
         };
+        // TODO: more than 3 smoothing group isn't handled.
```

Comment:
Let's properly support the case when we have more then 3 smooth groups. So you need to remove this and instead make change the NativeVertex to use SmallVec. You can use "const_new()" then. It's fine it NativeVertex loses it's copy propery and instead only implements clone. Smoothing groups should keep working as they are now, since they don't have the "-1" handled anywhere.